### PR TITLE
DRAFT: Pass for StridedSlice to Reshape

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
This is a draft for a pass that converts StridedSlice to Reshape.

For #7232

Signed-off-by: Hyun Sik Yoon <hyunsik.yoon@samsung.com>

### CODE REVIEW

File: compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ *                 |
+ *            [CircleNode]
+ *
+ * AFTER
+ *
+ *     [CircleNode]  [CircleConst]
+ *           \             /
+ *          [CircleReshape]
+ *                 |
+ *            [CircleNode]
```
Comment (seanshpark) [inline compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp:189]:
```
 *     [CircleNode]  [CircleConst]
 *           \             /
 *          [CircleReshape]            [CircleStridedSlice]
 *                 |
 *            [CircleNode]
```

`CircleStridedSlice` will exist as it is not deleted.

File: compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+} // namespace
+
+namespace luci
+{
+
+/**
+ * BEFORE
+ *
+ *     [CircleNode]  [CircleConst] for begin, end and strides, respectively
```
Comment (seanshpark) [inline compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp:177]: please remove `for begin, end and strides, respectively`

File: compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+ *          \             /
+ *          [CircleStridedSlice]
+ *                 |
+ *            [CircleNode]
+ *
+ * AFTER
+ *
+ *     [CircleNode]  [CircleConst] for shape
```
Comment (seanshpark) [inline compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp:185]: please remove `for shape`

File: compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+ *           \             /
+ *          [CircleReshape]
```
Comment (seanshpark) [inline compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp:187]: As I wrote before, this diagram is to show before-after the transformation and `[CircleStridedSlice]`  is not removed so it should be visible in the diagram.

Comment (seanshpark) [reply compiler/luci/pass/src/SubstituteStridedSliceToReshapePass.cpp:187]: If you don't underatand what I've wrote, please look at other Pass diagrams.

**Comment by hyunsik-yoon** [comment]
let me close this. all required commits were merged.

**Pull request closed without merging**